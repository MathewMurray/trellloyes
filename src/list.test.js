import List from './List';
import STORE from './STORE';
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

describe('smoke and snapshot tests',() =>{
it('generates list of cards',()=>{
const div = document.createElement('div');
ReactDOM.render(<List header='test' cards={[{id:"0", title:"test title", content:"test content"}]}/>,div)
ReactDOM.unmountComponentAtNode(div);
});
it('renders to the ui',() =>{
   const tree = renderer 
.create(<List header='test' cards={[{id:"0", title:"random words go here", content:"test content"}]} />)
   .toJSON();
   expect(tree).toMatchSnapshot();
});
});