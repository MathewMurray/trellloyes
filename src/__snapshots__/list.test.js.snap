// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke and snapshot tests renders to the ui 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      test
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        random words go here
      </h3>
      <p>
        test content
      </p>
    </div>
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
